Monorepo Beta 3 Releases (#2122)
ethereumjs/ethereumjs-monorepo: packages/trie/CHANGELOG.md

(modification: no type change headlines) and this project adheres to
[Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 5.0.0-beta.3 - 2022-08-10

Beta 3 release for the upcoming breaking release round on the [EthereumJS monorepo](https://github.com/ethereumjs/ethereumjs-monorepo) libraries, see the Beta 1 release notes for the main long change set description as well as the Beta 2 release notes for notes on some additional changes ([CHANGELOG](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/devp2p/CHANGELOG.md)).

### Root Hash Persistance

> The trie library now comes with a new constructor option `persistRoot` which is disabled by default but allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071), thanks to @faustbrian for the contribution! ❤️

Reviewer: ```suggestion
The trie library now comes with a new constructor option `persistRoot` which is disabled by default but allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071) and PR [#2123](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2123), thanks to @faustbrian for the contribution! ❤️
```

> The trie library now comes with a new constructor option `persistRoot` which is now enabled by default (**breaking**) and which allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071), thanks to @faustbrian for the contribution! ❤️

Reviewer 2: ```suggestion
The trie library now comes with a new constructor option `persistRoot` which is disabled by default but allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071), thanks to @faustbrian for the contribution! ❤️
```

> The trie library now comes with a new constructor option `persistRoot` which is now enabled by default (**breaking**) and which allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071), thanks to @faustbrian for the contribution! ❤️

Reviewer: ```suggestion
The trie library now comes with a new constructor option `persistRoot` which is now enabled by default (**breaking**) and which allows to persist state root updates along write operations directly in the DB and therefore omits the need to manually set to a new state root, see PR [#2071](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2071), thanks to [faustbrian](https://github.com/faustbrian) for the contribution! ❤️
```

> To switch back to the old behavior you can pass in the option set to `false` on instantiation:

Reviewer 2: We should remove this line now or else switch to the below
```suggestion
To switch to the new behavior you can pass in the option set to `true` on instantiation:
```

Reviewer: Same comment, this should be updated to `true` and the example below adjusted accordingly.

Author: Sorry for dismissing again, but I just do not have time to have the tests running once again and would want to have these releases still out on Thursday and not on a Friday. 😬

Author: Finally fixed this in a doc fixes/improvements PR I will push soon. 🙂

> To switch back to the old behavior you can pass in the option set to `false` on instantiation:

Reviewer 2: ```suggestion
To switch to the new behavior you can pass in the option set to `true` on instantiation:
```

Reviewer: Unless I'm misunderstanding, I don't think this is correct, since the new behavior (`true`) is default and therefore passing `true` has no effect.

Reviewer: Also the code below specifies `false`

Reviewer: Actually I wasn't aware of the latest iteration: `https://github.com/ethereumjs/ethereumjs-monorepo/pull/2123`. We should then update the text as you mentioned @acolytec3, update the example below, and add PR 2123 to the PR links above.

Author: 👍

```typescript
import { Trie, LevelDB } from '@ethereumjs/trie'
import { Level } from 'level'

> const trie = new Trie({ db: new LevelDB(new Level('MY_TRIE_DB_LOCATION')), persistRoot: false })

Reviewer: ```suggestion
const trie = new Trie({ db: new LevelDB(new Level('MY_TRIE_DB_LOCATION')), persistRoot: true })
```

> const trie = new Trie({ db: new LevelDB(new Level('MY_TRIE_DB_LOCATION'), persistRoot: false) })

Reviewer 2: ```suggestion
const trie = new Trie({ db: new LevelDB(new Level('MY_TRIE_DB_LOCATION'), persistRoot: true) })
```

> const trie = new Trie({ db: new LevelDB(new Level('MY_TRIE_DB_LOCATION'), persistRoot: false) })

Author: Have manually/locally constructed the correct version out of these two comments. 😋

```

### Other Changes

> - Fix: Pass down a custom hash function for hashing on trie copies, PR [#2068](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2068)

Author: I don't think this is necessary in the context of these Beta updates.
